src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -0,0 +1,412 @@
+        try:
+            arg = mapping[scalar_arg.intrinsic_type](
+                scalar_arg.name, interface=ArgumentInterface(
+                    INTENT_MAPPING[scalar_arg.intent]))
+        except KeyError:
+            raise NotImplementedError(
+                "scalar of type {0} not implemented"
+                "".format(scalar_arg.intrinsic_type))
+        self._symbol_table.add(arg)
+        self._arglist.append(arg)
+
+    def fs_common(self, function_space, var_accesses=None):
+        ''' Create LFRic Number of Dofs object '''
+
+        fs_name = function_space.orig_name
+        arg = NumberOfDofsDataSymbol("ndf_{0}".format(fs_name), fs_name,
+                                     interface=self.read_access)
+        self._symbol_table.add(arg)
+        self._arglist.append(arg)
+        self._ndofs_map[fs_name] = arg
+
+    def fs_intergrid(self, function_space, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("fs_integrated not implemented")
+
+    def fs_compulsory_field(self, function_space, var_accesses=None):
+        '''Create LFRic Number of Unique dofs and dofmap objects'''
+
+        fs_name = function_space.orig_name
+        arg = NumberOfUniqueDofsDataSymbol("undf_{0}".format(fs_name), fs_name,
+                                           interface=self.read_access)
+        self._symbol_table.add(arg)
+        self._arglist.append(arg)
+        self._undfs_map[fs_name] = arg
+
+        # The dimension argument may not have been declared yet so use
+        # an integer as a placeholder.
+        arg = DofMapDataSymbol("dofmap_{0}".format(fs_name), [1], fs_name,
+                               interface=self.read_access)
+        self._symbol_table.add(arg)
+        self._arglist.append(arg)
+        self._dofmaps_list.append(arg)
+
+    def banded_dofmap(self, function_space, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("banded_dofmap not implemented")
+
+    def indirection_dofmap(self, function_space, operator=None,
+                           var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("indirection_dofmap not implemented")
+
+    def basis(self, function_space, var_accesses=None):
+        ''' Implemented '''
+        from psyclone.dynamo0p3 import VALID_EVALUATOR_SHAPES, \
+            VALID_QUADRATURE_SHAPES
+        for shape in self._kern.eval_shapes:
+            if shape in VALID_QUADRATURE_SHAPES:
+                # A kernel stub won't have a name for the corresponding
+                # quadrature argument so we create one by appending the last
+                # part of the shape name to "qr_".
+                quad_name = shape.split("_")[-1]
+                basis_name = function_space.get_basis_name(
+                    qr_var="qr_"+quad_name)
+                fs_name = function_space.orig_name
+                arg = BasisFunctionDataSymbol(
```

Comment:
Perhaps, as suggested elsewhere, we need a sub-class of BasisFunctionDataSymbol for each quadrature shape?

Reply:
Yes, that's been done.